File: epicerie-vrac-projet-angular/src/app/models/commande.ts
import { Achat } from "./achat";
import { Facture } from "./facture";

export class Commande extends Achat {
    public numeroCommande: number = 0;
    public dateCommande: Date = new Date();
    // public client: Client = new Client();
    public facture: Facture = new Facture();
}

File: epicerie-vrac-projet/src/main/java/fr/formation/epicerievracprojet/models/Commande.java
package fr.formation.epicerievracprojet.models;

import java.time.LocalDate;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@NoArgsConstructor
@Getter
@Setter
@ToString(callSuper = true)
@Entity
public class Commande extends Achat {
	
	private int numeroCommande;
	
	private LocalDate dateCommande;
	
	@ManyToOne
	@ToString.Exclude
	@JsonIgnore
	private Client client;
	
	@OneToOne(mappedBy = "commande", cascade = CascadeType.ALL)
	private Facture facture;
	
}
